supervisor/dbus/network/setting/generate.py

```diff
@@ -93,10 +100,21 @@ def _get_ipv4_connection_settings(ipv4setting) -> dict:
     return ipv4
 
 
-def _get_ipv6_connection_settings(ipv6setting) -> dict:
+def _get_ipv6_connection_settings(ipv6setting: Ip6Setting | None) -> dict:
     ipv6 = {}
     if not ipv6setting or ipv6setting.method == InterfaceMethod.AUTO:
         ipv6[CONF_ATTR_IPV6_METHOD] = Variant("s", "auto")
+        if ipv6setting:
+            if ipv6setting.addr_gen_mode == InterfaceAddrGenMode.EUI64:
+                ipv6[CONF_ATTR_IPV6_ADDR_GEN_MODE] = Variant("i", 0)
+            else:
+                ipv6[CONF_ATTR_IPV6_ADDR_GEN_MODE] = Variant("i", 1)
+            if ipv6setting.ip6_privacy == InterfaceIp6Privacy.DISABLED:
+                ipv6[CONF_ATTR_IPV6_PRIVACY] = Variant("i", 0)
+            elif ipv6setting.ip6_privacy == InterfaceIp6Privacy.ENABLED_PREFER_PUBLIC:
+                ipv6[CONF_ATTR_IPV6_PRIVACY] = Variant("i", 1)
+            else:
+                ipv6[CONF_ATTR_IPV6_PRIVACY] = Variant("i", 2)
```

Comment:
Hm, we decided to use `link-local` IPv6 mode when the user selects disabled, do we want these settings maybe also apply then? :thinking: